gensim/models/ldamulticore.py

```diff
@@ -257,47 +258,45 @@ def update(self, corpus, chunks_as_numpy=False):
         def rho():
             return pow(self.offset + pass_ + (self.num_updates / self.chunksize), -self.decay)
 
+        def process_result_queue(force=False):
+            """
+            Clear the result queue, merging all intermediate results, and update the
+            LDA model if necessary.
+
+            """
+            merged_new = False
+            while not result_queue.empty():
+                other.merge(result_queue.get())
+                queue_size[0] -= 1
+                merged_new = True
+
+            if (force and merged_new and queue_size[0] == 0) or (other.numdocs >= updateafter):
```

Comment:
Check on `not self.batch` is redundant as if `self.batch` is truthy so `updateafter` is equal to `len(corpus)`. It means there will not be any temporary mstep, only after all processed docs or in `force` case. It's exactly mean what we want.